atomic_reactor/plugins/post_fetch_docker_archive.py

```diff
@@ -6,106 +6,45 @@
-        if self.method == 'gzip':
-            outfile = outfile.format('gz')
-            fp = gzip.open(outfile, 'wb', compresslevel=6)
-        elif self.method == 'lzma':
-            outfile = outfile.format('xz')
-            fp = lzma.open(outfile, 'wb')
-        else:
-            raise RuntimeError('Unsupported compression format {0}'.format(self.method))
-
-        _chunk_size = 1024**2  # 1 MB chunk size for reading/writing
-        self.log.info('compressing image %s to %s using %s method',
-                      self.workflow.image, outfile, self.method)
-        data = stream.read(_chunk_size)
-        while data != b'':
-            fp.write(data)
-            data = stream.read(_chunk_size)
-
-        self.uncompressed_size = stream.tell()
-
-        return outfile
-
     def run(self):
         if is_scratch_build(self.workflow):
             # required only to make an archive for Koji
             self.log.info('scratch build, skipping plugin')
             return
 
-        if self.load_exported_image and len(self.workflow.data.exported_image_sequence) > 0:
-            image_metadata = self.workflow.data.exported_image_sequence[-1]
-            image = image_metadata.get('path')
-            image_type = image_metadata.get('type')
-            self.log.info('preparing to compress image %s', image)
-            with open(image, 'rb') as image_stream:
-                outfile = self._compress_image_stream(image_stream)
-        else:
-            if self.source_build:
-                self.log.info('skipping, no exported source image to compress')
-                return
-            image = self.workflow.image
-            image_type = IMAGE_TYPE_DOCKER_ARCHIVE
-            self.log.info('fetching image %s from docker', image)
-            # OSBS2 TBD
-            # Set image_tarball_path, if image tarball is not yet
-            # stored somewhere, download it
-            # with imageutil.download_image_archive_tarball()
-            image_tarball_path = None
-            with open(image_tarball_path, 'rb') as image_tarball_file:
-                outfile = self._compress_image_stream(image_tarball_file)
+        if self.source_build:
+            self.log.info('skipping, no exported source image')
+            return
+        image = self.workflow.image
+        image_type = IMAGE_TYPE_DOCKER_ARCHIVE
+        self.log.info('fetching image %s from docker', image)
+        # OSBS2 TBD
+        # Set image_tarball_path, if image tarball is not yet
+        # stored somewhere, download it
+        # with imageutil.download_image_archive_tarball() for each platform
+        outfile = os.path.join(self.workflow.source.workdir,
+                               EXPORTED_COMPRESSED_IMAGE_NAME_TEMPLATE)
+        os.mknod(outfile)
         metadata = get_exported_image_metadata(outfile, image_type)
 
-        if self.uncompressed_size != 0:
-            metadata['uncompressed_size'] = self.uncompressed_size
```

Comment:
Not sure about removing `uncompressed_size`, doesn't some metadata plugin later use it?

Reply:
Nevermind, nothing uses it, not even in the osbs_ocp3 branch :raised_eyebrow: